packages/agent/src/agent/http/index.ts

```diff
@@ -219,9 +226,11 @@ export class HttpAgent implements Agent {
       const location = typeof window !== 'undefined' ? window.location : undefined;
       if (!location) {
         this._host = new URL('https://icp-api.io');
-        console.warn(
-          'Could not infer host from window.location, defaulting to mainnet gateway of https://icp-api.io. Please provide a host to the HttpAgent constructor to avoid this warning.',
-        );
+        this.log.notify({
+          level: 'warn',
+          message:
+            'Could not infer host from window.location, defaulting to mainnet gateway of https://icp-api.io. Please provide a host to the HttpAgent constructor to avoid this warning.',
+        });
```

Comment:
Suggestion to simplify the calling interface
```suggestion
        this.log.warn('Could not infer host from window.location, defaulting to mainnet gateway of https://icp-api.io. Please provide a host to the HttpAgent constructor to avoid this warning.',);
```

Reply:
Done, and I've also made `log` callable by having the class extend Function